File: tests/fixtures/cs/sample.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace MyApp.Models
{
    public interface IAnimal
    {
        string Name { get; }
        void Speak();
    }

    public abstract class BaseCreature : IAnimal
    {
        public string Name { get; set; }
        public abstract void Speak();
    }

    public class Dog : BaseCreature
    {
        public override void Speak()
        {
            Console.WriteLine("Woof");
        }

        public async Task<string> FetchAsync(string item)
        {
            await Task.Delay(100);
            return item;
        }
    }

    public struct Point
    {
        public double X;
        public double Y;
    }

    public enum Color
    {
        Red,
        Green,
        Blue
    }

    public static class MathHelper
    {
        public static int Add(int a, int b) => a + b;
    }
}

File: tests/fixtures/swift/sample.swift
import Foundation
import UIKit

protocol Animal {
    var name: String { get }
    func speak()
}

class Dog: Animal {
    var name: String

    init(name: String) {
        self.name = name
    }

    func speak() {
        print("Woof")
    }

    func fetch(item: String) -> String {
        return item
    }
}

struct Point {
    var x: Double
    var y: Double
}

enum Color {
    case red
    case green
    case blue
}

extension Dog {
    func describe() -> String {
        return "A dog named \(name)"
    }
}

public class ServiceManager {
    private var isActive: Bool = true

    public func start() {
        isActive = true
    }

    private func stop() {
        isActive = false
    }
}

typealias Handler = (String) -> Void
